ouroboros-network/test/Test/Mux.hs

```diff
@@ -390,61 +532,268 @@ prop_mux_starvation response0 response1 =
+
+        return (server_r, said)
+
+    -- Server that expects to receive a specific ByteString.
+    -- Doesn't send a reply.
+    serverRsp stopVar chan =
+        atomically (takeTMVar stopVar) >>= loop
+      where
+        loop e | e == BL.empty = return ()
+        loop e = do
+            msg_m <- recv chan
+            case msg_m of
+                 Just msg ->
+                     case BL.stripPrefix msg e of
+                          Just e' -> loop e'
+                          Nothing -> error "recv corruption"
+                 Nothing -> error "eof corruption"
+
+    writeSdu _ payload | payload == BL.empty = return ()
+    writeSdu queue payload = do
+        let (!frag, !rest) = BL.splitAt 0xffff payload
+            sdu' = Mx.MuxSDU (Mx.RemoteClockModel 0) (Mx.AppProtocolId ChainSyncSmall)
+                              Mx.ModeInitiator
+                             (fromIntegral $ BL.length frag) frag
+            !pkt = Mx.encodeMuxSDU (sdu' :: Mx.MuxSDU TestProtocolsSmall)
+
+        atomically $ writeTBQueue queue pkt
+        writeSdu queue rest
+
+    -- Unless we are in Larval or Connected we fake version negotiation before
+    -- we run the test.
+    {- Not yet! setup state q | state /= Mx.Larval && state /= Mx.Connected = do
+        let msg = Mx.MsgInitReq [version0]
+            blob = toLazyByteString $ Mx.encodeControlMsg msg
+            pkt = Mx.MuxSDU (Mx.RemoteClockModel 0) Mx.Muxcontrol Mx.ModeInitiator
+                            (fromIntegral $ BL.length blob) blob
+        atomically $ writeTBQueue q $ Mx.encodeMuxSDU (pkt :: Mx.MuxSDU TestProtocols1)
+        return () -}
+    setup _ _ = return ()
+
+    stateLabel (ArbitraryInvalidSDU _ state _) = "Invalid " ++ versionLabel state
+    stateLabel (ArbitraryValidSDU _ state _)   = "Valid " ++ versionLabel state
+
+    versionLabel Mx.Larval    = "before version negotiation"
+    versionLabel Mx.Connected = "before version negotiation"
+    versionLabel _            = "after version negotiation"
+
+    violationLabel (ArbitraryValidSDU _ _ err_m) = sduViolation err_m
+    violationLabel (ArbitraryInvalidSDU _ _ err) = sduViolation $ Just err
+
+    sduViolation (Just Mx.MuxUnknownMiniProtocol) = "unknown miniprotocol"
+    sduViolation (Just Mx.MuxDecodeError        ) = "decode error"
+    sduViolation (Just Mx.MuxIngressQueueOverRun) = "ingress queue overrun"
+    sduViolation (Just _                        ) = "unknown violation"
+    sduViolation Nothing                          = "none"
+
+prop_demux_sdu_sim :: ArbitrarySDU
+                     -> Property
+prop_demux_sdu_sim badSdu =
+    let r_e =  runSimStrictShutdown $ prop_demux_sdu badSdu in
+    case r_e of
+         Left  _ -> property False
+         Right r -> r
+
+prop_demux_sdu_io :: ArbitrarySDU
+                    -> Property
+prop_demux_sdu_io badSdu = ioProperty $ prop_demux_sdu badSdu
+
+demo :: forall m block.
```

Comment:
In other test cases we use `name` and `prop_name_ST` and `prop_name_IO`.  Let's pick the same convention.  `demo` -> `mux_demo`, `prop_mux_demo_sim` -> `prop_mux_demo_ST` ...

Reply:
It follows the same convention as the chainsync test in Socket.hs and in Pipe.hs which are both called `demo`.

Reply:
Tath's ok for now; it would be nice to have the same convention applied across different parts of the test suite.

Reply:
👍